packages/devextreme/js/__internal/ui/form/m_form.layout_manager.utils.ts

```diff
@@ -153,10 +197,48 @@ function _convertToEditorOptions({
   const stylingMode = externalEditorOptions?.stylingMode || editorStylingMode;
   const useSpecificLabelOptions = EDITORS_WITH_SPECIFIC_LABELS.includes(editorType);
 
+  let editorOptionsWithDropDown: Partial<dxDropDownEditorOptions<dxTextBox>> = {};
+  const useDropDownOptions = DROP_DOWN_EDITORS.includes(editorType);
+
+  if (useDropDownOptions) {
```

Comment:
[refactoring / code splitting]
I suggest you moving this new code to a separate method with an `editorType` parameter.
It'll simplify reading of the main function and will allow us to get of this condition

Reply:
moved